planner/core/logical_plan_builder.go

```diff
@@ -700,6 +700,17 @@ func (b *PlanBuilder) buildProjection4Union(u *LogicalUnionAll) {
 	// Process each child and add a projection above original child.
 	// So the schema of `UnionAll` can be the same with its children's.
 	for childID, child := range u.children {
+		uniqueIDToOutputIdxes := make(map[int][]int)
+		if p, ok := child.(*LogicalProjection); ok {
+			for i, expr := range p.Exprs {
+				col, ok1 := expr.(*expression.Column)
+				if !ok1 {
+					continue
+				}
+				uniqueID := int(col.UniqueID)
```

Comment:
Can we keep it as `int64`? Although it is very unlikely that it can overflow the int.